src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -454,10 +470,13 @@ function VisualizeAppController(
         $scope._handler.destroy();
       }
       savedVis.destroy();
-      stateMonitor.destroy();
       filterStateManager.destroy();
       subscriptions.unsubscribe();
       $scope.vis.off('apply', _applyVis);
+
+      unsubscribePersisted();
+      stateContainerSubscription.unsubscribe();
```

Comment:
This throws an error:
```
TypeError: stateContainerSubscription.unsubscribe is not a function
    at :5301/dqa/bundles/14.bundle.js:736
    at Scope.$broadcast (:5301/dqa/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:211591)
    at Scope.$destroy (:5301/dqa/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:211205)
    at cleanupLastView (webpack://%5Bname%5D/./node_modules/angular-route/angular-route.js?:1183)
    at eval (webpack://%5Bname%5D/./node_modules/angular-route/angular-route.js?:1216)
    at publicLinkFn (:5301/dqa/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:201567)
    at lazyCompilation (:5301/dqa/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:201982)
    at boundTranscludeFn (:5301/dqa/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:201746)
    at controllersBoundTransclude (:5301/dqa/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:202506)
    at update (webpack://%5Bname%5D/./node_modules/angular-route/angular-route.js?:1209)
(anonymous) @ :5301/dqa/bundles/kbn-ui-shared-deps/kbn-ui-shared-deps.js:207438
```

looks like `stateContainer.subscribe` is not returning a `Subscription` object, but a simple function

Reply:
Just for info: 

That `.subscribe` there, is to make state containers redux interface compatible. And that is why it returns an unsubscribe function. 

To use a proper observable you could use:

```
const state$ = container.$state;
```

Reply:
fixed